src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -0,0 +1,454 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
+{
+    public class HttpRequestStreamTests : IDisposable
+    {
+        private HttpListenerFactory _factory;
+        private HttpListener _listener;
+        private GetContextHelper _helper;
+
+        public HttpRequestStreamTests()
+        {
+            _factory = new HttpListenerFactory();
+            _listener = _factory.GetListener();
+            if (!PlatformDetection.IsWindows7)
```

Comment:
ClientWebSocket isn't supported in Windows 7. I have updated the PR and this whole code block and reliance on ClientWebSocket has gone.